### ISSUE TITLE
reportUnknownMemberType thrown for re-exported types but not directly exported types

### ISSUE THREAD

**Issue by brandon-leapyear** [microsoft/pyright#3334]
**Describe the bug**
Calling a partially-unknown function after directly importing it (`from .. import ..`) doesn't error, but calling it with the module namespace errors.

**To Reproduce**
```python
# foo.py
def test(x: list) -> None:
    print(x)
```
```python
# test1.py
from foo import test
test([])
```
```python
# test2.py
import foo
foo.test([])
```
```json
{
  "include": ["test*.py"],
  "typeCheckingMode": "strict"
}
```
Running pyright with this config on this project will show an error in `test2.py` but not in `test1.py`:
```
test2.py:2:1 - error: Type of "test" is partially unknown
    Type of "test" is "(x: list[Unknown]) -> None" (reportUnknownMemberType)
```

**Expected behavior**
I would expect the error to be the same regardless of how a function is imported.

**Screenshots or Code**

**VS Code extension or command-line**
CLI (fresh `npm install pyright`)

**Additional context**

**Comment by erictraut** [comment]
The `reportUnknownMemberType` diagnostic check specifically affects member access expressions of the form `a.b` (or in your example `foo.test`).

But you raise a good point in that this does leave a potential hole in strict type check. Maybe this case should be covered in the `reportUnknownVariableType` check. I'll need to think about whether it would be better to report that error when the symbol is imported (e.g. in the `from foo import test` statement) or when it is used (e.g. in the `test([])` statement).

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.237, which I just published. It will also be included in the next release of pylance.

**Comment by brandon-leapyear** [comment]
That was fast! Thank you!
